File: FrontEnd/xego_driver/lib/models/Dto/registration_request_dto.dart
class RegistrationRequestDto {
  String email;
  String userName;
  String phoneNumber;
  String password;
  String firstName;
  String lastName;
  String address;
  String role;

  RegistrationRequestDto({
    required this.email,
    required this.userName,
    required this.phoneNumber,
    required this.password,
    required this.firstName,
    required this.lastName,
    required this.address,
    required this.role,
  });

  factory RegistrationRequestDto.fromJson(Map<String, dynamic> json) {
    return RegistrationRequestDto(
      email: json['Email'],
      userName: json['UserName'],
      phoneNumber: json['PhoneNumber'],
      password: json['Password'],
      firstName: json['FirstName'],
      lastName: json['LastName'],
      address: json['Address'],
      role: json['Role'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'Email': email,
      'UserName': userName,
      'PhoneNumber': phoneNumber,
      'Password': password,
      'FirstName': firstName,
      'LastName': lastName,
      'Address': address,
      'Role': role,
    };
  }
}

File: BackEnd/XeGo-Microservices/Services/Auth/XeGo.Services.Auth.API/Models/Dto/RegistrationRequestDto.cs
using System.ComponentModel.DataAnnotations;

namespace XeGo.Services.Auth.API.Models.Dto
{
    public class RegistrationRequestDto
    {
        public string? Email { get; set; }
        [Required] public string UserName { get; set; } = string.Empty!;
        [Required] public string PhoneNumber { get; set; } = string.Empty!;
        [Required] public string Password { get; set; } = string.Empty!;
        [Required] public string FirstName { get; set; } = string.Empty!;
        [Required] public string LastName { get; set; } = string.Empty!;
        [Required] public string Address { get; set; } = string.Empty!;
        [Required] public string Role { get; set; } = string.Empty!;

    }
}
